NGitLab/IRepositoryClient.cs

```diff
@@ -21,7 +21,7 @@ public interface IRepositoryClient
 
     void GetRawBlob(string sha, Action<Stream> parser);
 
-    void GetArchive(Action<Stream> parser);
+    void GetArchive(Action<Stream> parser, string sha = null, string format = null);
```

Comment:
This is a breaking change.

Furthermore, we may eventually want to add support for a subpath specification (see [here](https://docs.gitlab.com/ee/api/repositories.html#get-file-archive)).

So I suggest the following:

1. Introduce a `FileArchiveQuery` type:
    ```cs
    public sealed class FileArchiveQuery
    {
        public Sha1 Sha { get; set; }
        public ArchiveFormat Format { get; set; }
    }
    ```
    where `ArchiveFormat` would be an `enum` of the available format options
1. Add a `GetArchive(Action<Stream> parser, FileArchiveQuery query)` overload

This way, the API will be more future-proof.

Reply:
Thanks for the detailed feedback @louis-z. I will make the requested changes and push an update for you to review.

Reply:
@louis-z - I have made the requested changes but am unsure if there is a preferred way of handling the file extension translation. Please see my comments below regarding this.